src/workloads/execution/Densify.yml

```diff
@@ -0,0 +1,421 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This workload test tests the performance of the $densify stage. $densify can operate on both dates
+  and numbers.
+
+Actors:
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+  - Repeat: 1
+    Database: &db test
+    Threads: 1
+    CollectionCount: 1
+    DocumentCount: 10000
+    BatchSize: &batchSize 1000
+    Document:
+      partitionKey: {^RandomInt: {min: 1, max: 100}}
+      # Test a subset of units that have different characteristics.
+      number: {^RandomInt: {min: 0, max: 20000}} # Seperate codepath for numeric DensifyValues.
+      milliseconds: {^RandomDate: {min: "2021-01-01T00:00:00.000Z", max: "2021-01-01T00:00:20.000Z"}} # Most basic date unit.
+      hours: {^RandomDate: {min: "2021-01-01", max: "2023-01-01"}} # Date unit with constant number of milliseconds.
+      months: {^RandomDate: {min: "2021-01-01", max: "4000-01-01"}} # Date unit with non-constant number of milliseconds.
```

Comment:
Question: what is the difference between `hours` and `months` in terms of constant/non-constant number of milliseconds? What I'm seeing here is that the `months` values will just be spread across a wider domain.

Reply:
An hour will always be the same number of milliseconds (1000*60*60), but a month is a different number of milliseconds depending on the number of days in that month, which needs to be taken into account when determining whether a given value is on the step defined in the stage. There's different codepaths in the densify stage to account for this, and there's some potential for inefficiencies in the case for months, quarters and years, so I wanted to record metrics for both months and hours to find any large discrepancies.

This could certainly be clearer from the comment, and I'll add more details!

Reply:
Ah got it, thanks for the explanation! An edited comment would be great.

Reply:
done.